### PR TITLE
Add HackerOne Icon

### PR DESCRIPTION
**Pull request by MiahaCybersec**
Adds 1 new icon: `HackerOne`

In the event that a security researcher uses Starlight for their personal blog or website, it might be nice for them to be able to link their HackerOne profile at the top of their website.

### CODE REVIEW

File: packages/starlight/components/Icons.ts [hunk 1]
```diff
@@ -109,6 +109,7 @@ export const BuiltInIcons = {
...
 	matrix:
 		'<path d="M22.5 1.5v21h-2.25V24H24V0h-3.75v1.5h2.25ZM7.46 7.95V9.1h.04a3.02 3.02 0 0 1 2.61-1.39c.54 0 1.03.1 1.48.32.44.2.78.58 1.01 1.1.26-.37.6-.7 1.03-.99.44-.28.95-.43 1.55-.43.45 0 .87.06 1.26.17.38.11.71.29.99.53.27.24.49.56.64.95.15.4.23.86.23 1.42v5.72h-2.34v-4.85c0-.29-.01-.56-.04-.8a1.73 1.73 0 0 0-.18-.67 1.1 1.1 0 0 0-.44-.45 1.6 1.6 0 0 0-.78-.16c-.33 0-.6.06-.8.19-.2.12-.37.29-.48.5a2 2 0 0 0-.23.69c-.04.26-.06.52-.06.78v4.77H10.6v-4.8l-.01-.75a2.29 2.29 0 0 0-.14-.69c-.08-.2-.23-.38-.42-.5a1.5 1.5 0 0 0-.85-.2c-.15.01-.3.04-.44.08-.19.06-.37.15-.52.28-.18.14-.32.34-.44.6-.12.26-.18.6-.18 1.02v4.96H5.25V7.94h2.21ZM1.5 1.5v21h2.25V24H0V0h3.75v1.5H1.5Z"/>',
+	hackerone: '<path d="M7.207 0q-.725.002-1.182.3c-.305.2-.46.463-.46.78v21.809q0 .414.476.76.472.351 1.166.351.663 0 1.168-.35.504-.346.506-.761V1.082c0-.32-.163-.577-.49-.782Q7.901.002 7.207 0m9.523 8.662q-.724-.001-1.168.3l-4.439 2.783c-.199.186-.284.469-.247.855q.052.575.524 1.09c.314.347.666.563 1.068.655q.596.134.896-.143l1.755-1.095v9.782q0 .414.461.76c.3.234.687.351 1.15.351q.695 0 1.199-.35c.337-.233.506-.484.506-.761V9.739q-.001-.48-.49-.776-.488-.303-1.215-.301"/>',
```
Comment (delucis) [inline packages/starlight/components/Icons.ts:112]: Ran this through [SVGOMG](https://jakearchibald.github.io/svgomg/) to reduce it’s size a little:

```suggestion
	hackerone: '<path d="M7.2 0Q6.5 0 6 .3a1 1 0 0 0-.4.8v21.8q0 .4.4.7.5.4 1.2.4t1.2-.3q.5-.4.5-.8V1c0-.3-.2-.6-.5-.8Q7.9 0 7.2 0m9.5 8.7q-.7 0-1.1.3L11 11.7c-.2.2-.3.5-.2.9q0 .6.5 1c.3.4.7.7 1 .7q.7.2 1-.1l1.7-1v9.7q0 .4.5.7c.3.3.7.4 1.1.4q.7 0 1.2-.3c.4-.3.5-.5.5-.8V9.7q0-.4-.5-.7-.4-.3-1.2-.3"/>',
```

**Review by delucis — COMMENTED** [review]
Thanks for the contribution @MiahaCybersec — looks good, I think.

One detail: what do you think of using `hackerOne` as the icon/config key instead of the all lowercase `hackerone`? In general we do that for icons that are technically two words (e.g. `openCollective`.

File: packages/starlight/components/Icons.ts (hunk 1 above)
Comment (MiahaCybersec) [reply packages/starlight/components/Icons.ts]: What is generally considered best practice when adding an SVG to this repo? Number precision of 1 and multiple passes on SVGOMG?

**Review by delucis — APPROVED** [review]
Thanks for making those updates @MiahaCybersec! Looks ready to go out with our next release 🚀

**Pull request merged**